input_tool.py

```diff
@@ -0,0 +1,203 @@
+                observation_name VARCHAR,
+                variable_id INTEGER NOT NULL,
+                CONSTRAINT fk_variable
+                    FOREIGN KEY (variable_id)
+                        REFERENCES variables(variable_id)
+                        ON DELETE CASCADE      
+    )""")
+    # Plots table
+    cur.execute("""CREATE TABLE IF NOT EXISTS plots (
+                plot_id serial PRIMARY KEY,
+                div VARCHAR,
+                script VARCHAR,
+                experiment_id INTEGER NOT NULL,
+                group_id INTEGER NOT NULL,
+                observation_id INTEGER NOT NULL,
+                CONSTRAINT fk_experiment
+                    FOREIGN KEY (experiment_id)
+                        REFERENCES experiments(experiment_id)
+                        ON DELETE CASCADE,
+                CONSTRAINT fk_group
+                    FOREIGN KEY (group_id)
+                        REFERENCES groups(group_id)
+                        ON DELETE CASCADE,
+                CONSTRAINT fk_observation
+                    FOREIGN KEY (observation_id)
+                        REFERENCES observations(observation_id)
+                        ON DELETE CASCADE
+    )""")
+
+def drop_tables(cur):
+    cur.execute("DROP TABLE IF EXISTS owners CASCADE")
+    cur.execute("DROP TABLE IF EXISTS experiments CASCADE")
+    cur.execute("DROP TABLE IF EXISTS plots CASCADE")
+    cur.execute("DROP TABLE IF EXISTS groups CASCADE")
+    cur.execute("DROP TABLE IF EXISTS observations CASCADE")
+    cur.execute("DROP TABLE IF EXISTS variables CASCADE")
+
+def load_dataset_to_db(cur):
+    with open("dataset.json", 'rb') as dataset:
+        sample_dataset = json.load(dataset)
+    for owner in sample_dataset['owners']:
+        add_user(cur, owner)
+    for experiment in sample_dataset['experiments']:
+        add_experiment(cur, experiment)
+    for plot in (sample_dataset['plots']):
+        add_plot(cur, plot, sample_dataset['observation_dirs'])
+
+def get_observation_name(obs_dirs, filename):
+    for observation in obs_dirs:
+        obs_dir_path = os.path.join(EXPERIMENT_DATA_PATH, observation)
+        for plot in os.listdir(obs_dir_path):
+            if filename == plot:
+                return observation
+    return None
+
+def add_user(cur, user_obj):
+    # check preconditions
+    if 'owner_id' not in user_obj or 'username' not in user_obj:
+        print("Skipping current user: needs both owner_id and username")
+        return 1
+    elif 'first_name' not in user_obj or 'last_name' not in user_obj:
+        cur.execute("INSERT INTO owners (owner_id, username, first_name, last_name) VALUES (%s, %s, %s, %s)", (user_obj["owner_id"], user_obj["username"], user_obj["first_name"], user_obj["last_name"]))
+    else:
+        cur.execute("INSERT INTO owners (owner_id, username, first_name, last_name) VALUES (%s, %s, %s, %s)", (user_obj["owner_id"], user_obj["username"], user_obj["first_name"], user_obj["last_name"]))
```

Comment:
Why are these commands identical?

Reply:
In today's meeting we agreed to ignore first/last name for now (allow NULL), but to enforce not-null values for experiment, group, variable, and observation names.